### DISCUSSION TITLE
HUAWEI nova 11 connection problem

### DISCUSSION THREAD

**Discussion by Proxeny** [rustdesk/rustdesk#5599] [General]
Hello!

I have HUAWEI nova 11 (EMUI: 13.0.0 (13.0.0.136 (C10E2R1P2).
Nova freezes rustdesk connection:
1. If using rustdesk.apk (1.2.2) on the phone. Rust can not complete connection. If NOVA is connected via wifi it has the same effect on rustdesk.apk - I can not connect to an any client.;
2. If using Nova as a hotspot, Rustdesk desktop addition connects to a client, than loses connection. After reconnection - only freezed screenshot of client. (I've tried different laptops (HUAWEI, HP, Lenovo) and OSes (Win10, Win11, Xubuntu) - doesn't matter)
It doesn't matter if I use free server of my self hosted.  The same with any versions of RustDesk.
With the same sim card on an iPhone everything works perfect, even on XIAOMI Redmi 9.
I've tried to use different DNS, dropped network setting on the device etc, etc. So, it means, that the problem is in NOVA 11, not in Rustdesk.
I've opened an issue in HUAWEI support more than two weeks ago and they don't know how to fix it.
If anybody had any idea how to solve this issue, please, help.

I've got used to use RustDesk, it works everywhere brilliant, but not on HUAWEI.

**Comment by rustdesk** [comment]
Maybe we need an apk built with HarmonyOS sdk.

**Reply by Proxeny** [reply]
Hm, but this is not Harmony OS.
And if Nova works as a hotspot it doesn't matter what apk is installed. It seems to me that it is something in HUAWEI's firewall or something like this. They didn't give any useful advice still... (

**Reply by rustdesk** [reply]
It is abnormal if they can give you useful advice.
